includes/rest-api.php

```diff
@@ -190,9 +209,71 @@ function get_pull_content_list_args() {
 		),
 		'post_type'      => array(
 			'description'       => esc_html__( 'Limit results to content matching a certain type.', 'distributor' ),
-			'type'              => 'string',
-			'default'           => 'post',
-			'sanitize_callback' => 'sanitize_text_field',
+			'type'              => array( 'array', 'string' ),
+			'items'             => array(
+				'type' => 'string',
+			),
+			'default'           => array( 'post' ),
+			'validate_callback' => function( $param ) {
+				if ( is_string( $param ) ) {
+					return sanitize_key( $param ) === $param;
+				}
+
+				foreach ( $param as $post_type ) {
+					if ( sanitize_key( $post_type ) !== $post_type ) {
+						return false;
+					}
+				}
+
+				return true;
+			},
+			'sanitize_callback' => function( $param ) {
+				if ( is_string( $param ) ) {
+					$param = array( $param );
+				}
+
+				$allowed_post_types = array_keys(
+					get_post_types(
+						array(
+							'show_in_rest' => true,
+						)
+					)
+				);
+
+				/*
+				 * Only post types viewable on the front end should be allowed.
+				 *
+				 * Some post types may be visible in the REST API but not intended
+				 * to be viewed on the front end. This removes any such posts from the
+				 * list of allowed post types.
+				 *
+				 * `is_post_type_viewable()` is used to filter the results as
+				 * WordPress applies different rules for custom and built in post
+				 * types to determine whether they are viewable on the front end.
+				 */
+				$allowed_post_types = array_filter( $allowed_post_types, 'is_post_type_viewable' );
+
+				if ( in_array( 'any', $param, true ) ) {
+					$param = $allowed_post_types;
+				} else {
+					$param = array_intersect( $param, $allowed_post_types );
+				}
+
+				$param = array_filter(
+					$param,
+					function( $post_type ) {
+						$post_type_object = get_post_type_object( $post_type );
+						return current_user_can( $post_type_object->cap->edit_posts );
+					}
+				);
+
+				if ( empty( $param ) ) {
+					// This will cause the parameter to fall back to the default.
+					$param = null;
+				}
+
+				return $param;
+			},
 			'validate_callback' => 'rest_validate_request_arg',
```

Comment:
@peterwilsoncc This is a duplicate entry. Can we remove one?

Reply:
62d165fe17ce35010d197ebdd4e076800f96a311

Reply:
Good catch, thank you! Removed in 0efefdad961cb60de88776c43af3046c81065362